packages/orbit-components/src/Popover/components/ContentWrapper.js

```diff
@@ -216,6 +217,15 @@ const PopoverContentWrapper = ({
     [actions],
   );
 
+  const handleClick = React.useCallback(
+    ev => {
+      if (onClose && overlay.current.contains(ev.target)) {
```

Comment:
I forgot to add that after attaching `onMouseDown` to `StyledOverlay`, this check is now unnecessary because clicks outside of that element won't trigger this handler.